client/signup/navigation-link/index.jsx

```diff
@@ -53,10 +53,17 @@ export class NavigationLink extends Component {
 	 */
 	getPreviousStep() {
 		const { flowName, signupProgress, stepName } = this.props;
+		const steps = getFilteredSteps( flowName, signupProgress );
 
-		let steps = getFilteredSteps( flowName, signupProgress );
-		steps = steps.slice( 0, findIndex( steps, step => step.stepName === stepName ) );
-		const previousStep = findLast( steps, step => ! step.wasSkipped && step.stepName !== stepName );
+		if ( ! isArray( steps ) ) {
+			return { stepName: null };
```

Comment:
Why would that not return an array though? It seems like this check would conceal a legitimate bug if it were to happen one day

Reply:
Yeah, we could have ended up returning something else from `getFilteredSteps` and have been returning `{ stepName: null }` constantly which would have been tricky to debug - good catch!
I've removed this check 👍